### PR TITLE
[DRAFT] Onert-micro training PoCv2

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This draft introduces second version of the onert-micro training runtime. This is for one-stage training (without generating backprop graph part). Also this draft introduces weight divider tool.

for issue:  #12873

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Comment by chunseoklee** [comment]
cc @ljwoo94

File: onert-micro/eval-driver/TrainingDriver.cpp [hunk 1]
```diff
@@ -100,118 +92,180 @@ int entry(int argc, char **argv)
...
+  const char *input_target_test_data_path = argv[7];
+  const int32_t num_train_data_samples = atoi(argv[8]);
+  const int32_t num_test_data_samples = atoi(argv[9]);
+
+  DataBuffer circle_model = readFile(circle_model_path);
+  DataBuffer wof_data = readFile(wof_file_path);
+
+  // Configure training mode
+  onert_micro::OMConfig config;
+  config.wof_ptr = wof_data.data();
```
Comment (chunseoklee) [inline onert-micro/eval-driver/TrainingDriver.cpp:135]:
```suggestion
  config.wof_ptr = nullptr;
```
For june release, let's train circle-with-weight format.

Comment (BalyshevArtem) [reply onert-micro/eval-driver/TrainingDriver.cpp:135]: Then we need to figure out how to save the training result

File: onert-micro/eval-driver/TrainingDriver.cpp [hunk 2]
```diff
@@ -100,117 +92,236 @@ int entry(int argc, char **argv)
...
-  char *test_data = new char[input_size * num_test_data_samples];
-  char *label_test_data = new char[output_size * num_test_data_samples];
+  // Close file
+  out_file.close();
 
-  readDataFromFile(input_train_data_path, train_data, input_size * num_train_data_samples);
-  readDataFromFile(input_test_data_path, test_data, input_size * num_test_data_samples);
+  // Saving done
+  return true;
+}
```
Comment (BalyshevArtem) [inline onert-micro/eval-driver/TrainingDriver.cpp:117]: @chunseoklee, Introduce variant without wof file, only circle. In this variant:
We open circle file, we copy model data into RAM memory, train model (change our local model data), and then save it with current help method.

What do you think about this approach?

File: onert-micro/eval-driver/TrainingDriver.cpp (hunk 1 above)
Comment (BalyshevArtem) [reply onert-micro/eval-driver/TrainingDriver.cpp]: Introduced  #13107 (without weight divider), please check

File: onert-micro/eval-driver/TrainingDriver.cpp (hunk 2 above)
Comment (BalyshevArtem) [reply onert-micro/eval-driver/TrainingDriver.cpp:117]: Introduced  #13107 (without weight divider), please check

File: onert-micro/eval-driver/TrainingDriver.cpp (hunk 1 above)
Comment (chunseoklee) [reply onert-micro/eval-driver/TrainingDriver.cpp]: > Then we need to figure out how to save the training result

As you already implement at next PR, final weight should be saved as single circle file. Intermediate checkpoint keeps weight and other ongoing state at another file.

File: onert-micro/eval-driver/TrainingDriver.cpp (hunk 2 above)
Comment (chunseoklee) [reply onert-micro/eval-driver/TrainingDriver.cpp:117]: > We open circle file, we copy model data into RAM memory, train model (change our local model data), and then save it with current help method.

w/o checkpoint, this approach is exactly what I have in mind. 

After resuming training from checkpoint, we load weight from checkpoint(all or some), train with them, and then save it like : copy circle into RAM and copy current weight into circle buffer and dump file.

**Pull request closed without merging**